packages/block-library/src/navigation/index.php

```diff
@@ -117,7 +104,23 @@ private static function is_interactive( $attributes, $inner_blocks ) {
 	 * @return bool Returns whether or not a block needs a list item wrapper.
 	 */
 	private static function does_block_need_a_list_item_wrapper( $block ) {
-		return in_array( $block->name, static::$needs_list_item_wrapper, true );
+
+		/**
+		 * Filter the list of blocks that need a list item wrapper.
+		 *
+		 * Affords the ability to customize which blocks need a list item wrapper when rendered
+		 * within a core/navigation block.
+		 * This is useful for blocks that are not list items but should be wrapped in a list
+		 * item when used as a child of a navigation block.
+		 *
+		 * @since 6.5.0
+		 *
+		 * @param array $needs_list_item_wrapper The list of blocks that need a list item wrapper.
+		 * @return array The list of blocks that need a list item wrapper.
+		 */
+		$needs_list_item_wrapper = apply_filters( 'block_core_navigation_blocks_requiring_list_item_wrapper', static::$needs_list_item_wrapper );
```

Comment:
Let's try a shorter name for the filter:

```suggestion
		$needs_list_item_wrapper = apply_filters( 'block_core_navigation_listable', static::$needs_list_item_wrapper );
```

Reply:
What do you think about `block_core_navigation_listable_blocks` to make it more clear that you are filtering a block list?

Reply:
If the filter is well documented then we should be good with something like that. There's nothing to say that the filter name itself has to be able to perfectly communicate all things about the filter.

Reply:
I love `block_core_navigation_listable_blocks`

Reply:
I updated the name of the filter  👍 Thanks all!